python-packages/order_utils/src/zero_ex/order_utils/__init__.py

```diff
@@ -9,3 +9,132 @@
 --networkId 50 -m "concert load couple harbor equip island argue ramp clarify
 fence smart topic"``.
 """
+
+from mypy_extensions import TypedDict
+
+from eth_utils import is_address, keccak, to_checksum_address, to_bytes
+from web3 import Web3
+from web3.utils import datatypes
+import web3.exceptions
+
+
+NULL_ADDRESS = "0x0000000000000000000000000000000000000000"
+
+
+class Order(TypedDict):  # pylint: disable=too-many-instance-attributes
+    """Object representation of a 0x order."""
+
+    maker_address: str
+    taker_address: str
+    fee_recipient_address: str
+    sender_address: str
+    maker_asset_amount: int
+    taker_asset_amount: int
+    maker_fee: int
+    taker_fee: int
+    expiration_time_seconds: int
+    salt: int
+    maker_asset_data: str
+    taker_asset_data: str
+    exchange_address: str
+
+
+def make_empty_order() -> Order:
+    """Construct an empty order."""
+    return {
+        "maker_address": NULL_ADDRESS,
+        "taker_address": NULL_ADDRESS,
+        "sender_address": NULL_ADDRESS,
+        "fee_recipient_address": NULL_ADDRESS,
+        "maker_asset_data": NULL_ADDRESS,
+        "taker_asset_data": NULL_ADDRESS,
+        "exchange_address": NULL_ADDRESS,
+        "salt": 0,
+        "maker_fee": 0,
+        "taker_fee": 0,
+        "maker_asset_amount": 0,
+        "taker_asset_amount": 0,
+        "expiration_time_seconds": 0,
+    }
+
+
+def generate_order_hash_hex(order: Order) -> str:
+    # docstring considered all one line by pylint: disable=line-too-long
+    """Calculate the hash of the given order as a hexadecimal string.
+
+    >>> generate_order_hash_hex(
+    ...     {
+    ...         'maker_address': "0x0000000000000000000000000000000000000000",
+    ...         'taker_address': "0x0000000000000000000000000000000000000000",
+    ...         'fee_recipient_address': "0x0000000000000000000000000000000000000000",
+    ...         'sender_address': "0x0000000000000000000000000000000000000000",
+    ...         'maker_asset_amount': 1000000000000000000,
+    ...         'taker_asset_amount': 1000000000000000000,
+    ...         'maker_fee': 0,
+    ...         'taker_fee': 0,
+    ...         'expiration_time_seconds': 12345,
+    ...         'salt': 12345,
+    ...         'maker_asset_data': "0000000000000000000000000000000000000000",
+    ...         'taker_asset_data': "0000000000000000000000000000000000000000",
+    ...         'exchange_address': "0x0000000000000000000000000000000000000000",
+    ...     },
+    ... )
+    '55eaa6ec02f3224d30873577e9ddd069a288c16d6fb407210eecbc501fa76692'
```

Comment:
Note that this expected value comes from the existing test case in `packages/utils/test/sign_typed_data_utils_test.ts`.